src/kernel/context.cpp

```diff
@@ -10,13 +10,16 @@
 
 #include <string>
 
-
 namespace kernel {
 Context::Context()
 {
-    std::string sha256_algo = SHA256AutoDetect();
-    LogPrintf("Using the '%s' SHA256 implementation\n", sha256_algo);
-    RandomInit();
+    static bool initialized{false};
```

Comment:
nit: since the flag is about the globals, and not the context, being initialized - would `globals_initialized` be a better name? It doesn't make a difference now, but it will when the constructor does more than just setting up globals.

```suggestion
    static bool globals_initialized{false};
```